### PR TITLE
feat(tools/g1): port g1_slam_map_chunks_compaction_envelope lookup pair (refs #358)

### PR DESCRIPTION
**Pull request by cagataycali**
## What this ships

Ports the neon `_SlamRunner._process_frame` compaction trigger (`cagataycali/neon-the-g1/tools/g1_slam.py::_SlamRunner._process_frame` reads `if len(self._map_chunks) > 100:` and steals the chunk list under its own lock on strict-greater, stitches every chunk into a single `(N, 4)` XYZI array, calls the module-level `_voxel_dedup` helper, and installs the deduped result back as the sole chunk) as a read-only agent-facing lookup pair in `strands_robots.tools.g1.g1_slam_map_chunks_compaction_envelope`.

The neon runner reads a **strict-greater ceiling on the chunk list length**: while the runner's `_accumulating` flag is true, every admitted ICP frame appends one `(N, 4)` world-point chunk. The 101st append fires the steal-and-dedup pass on the `_process_frame` thread and collapses the chunk list back to a single deduped chunk. This is a **policy** -- amortise the compaction cost over ~10 seconds of accumulation at the observed 10 Hz LiDAR rate -- not a hardware limit.

## API

- `g1_list_slam_map_chunks_compaction_envelope()` -> `{envelope: {map_chunks_compaction_max: 100}, refusals: [{text: <module-local>}]}`
- `g1_slam_map_chunks_compaction_admits(batch_depth: int = 100)` -> `{admits: bool, refusals: [...], envelope: {...}}`

## Third twin on the same `_SlamRunner._process_frame` code path

The neon `_SlamRunner._process_frame` maintains three distinct in-memory bookkeeping ceilings on the same code path, each with a different remedy for a different surface:

- `g1_slam_pose_history_envelope` (the merged #3026) -- bookkeeping ceiling on the runner's own append log (`_pose_history`, remedy: shorten the session or buffer to disk)
- `g1_slam_frame_queue_envelope` (#3027, in flight) -- capacity policy on the producer-side channel (`_frame_q`, remedy: accept the freshness-preferring drop policy or throttle upstream)
- **this PR** -- compaction trigger on the accumulated world-point log (`_map_chunks`, remedy: shorten the accumulation window or install a compaction timer upstream)

They stay separate because a caller planning a runner deployment reads them on different arguments and with different remedies. Colocating would hand an agent planner a single refusal payload mixing three disjoint remedies against three distinct write paths.

## Also a twin of the merged #3014 (voxel_dedup) on the same compaction pass

- `g1_slam_voxel_dedup_envelope` (the merged #3014) names the 5 cm **cell size** the compaction reads at each fire
- **this PR** names the 100-chunk **trigger count** that decides *when* the compaction fires at all

A widen to one would not co-widen the other -- a coarser cell would not change the batch size, and a larger batch would not change the cell size -- so keeping them on separate surfaces keeps each widen local.

## Refusal strings

One module-local `_REFUSAL_TEXT` cites `strands-labs/robots#358` for both the over-ceiling rejection and the shared-domain shape mistake, so a misread of either grade surfaces the same remedy string on the same surface -- consistent with the twin envelopes.

## Why not borrow `7404`?

The motion-FSM `7404` entry (`Invalid FSM id - need FSM in {500, 501, 801}`) is the nearest neighbour but points at locomotion FSM transitions -- the wrong remedy for a bookkeeping compaction trigger on an in-memory Python list. Module-local text keeps the refusal on the same surface a future driver-side SLAM accumulation wrapper will land on.

## Import hygiene

`import strands_robots.tools.g1.g1_slam_map_chunks_compaction_envelope` pulls **no** `unitree_sdk2py`, `numpy`, `open3d`, `kiss_icp`, or stdlib `queue` submodule at load time -- the contract every other file in `strands_robots/tools/g1/` carries. Asserted in the test module's `TestTheImportPullsNoOptionalSlamModule` cell (three separate assertions: unitree, open3d/kiss_icp, and stdlib `queue`).

## Tests

**24 tests, 14.08s wall, no bus, no SDK.**

- 3 import-hygiene tests (no unitree/open3d/kiss_icp/queue leak)
- 4 snapshot-fidelity tests (constant type, neon value 100, list-verb shape, list/admits parity)
- 4 admits-at-ceiling boundary cases (100, 50, 1, default-zero-arg)
- 2 refuses-above-ceiling boundary cases (101, 36000 far-above)
- 8 shared-domain shape refusals (0, -1, True, False, non-int parametrized: `1.0, 100.0, "100", None`)
- 3 refusal-text-shared invariants

## Gates

- `ruff check` -- All checks passed!
- `ruff format --check` -- 2 files already formatted
- `mypy strands_robots/tools/g1/g1_slam_map_chunks_compaction_envelope.py` -- 0 errors on touched file (pre-existing errors in `mesh/core.py` and `simulation/mujoco/rendering.py` are unrelated)
- `pytest tests/drivers/test_g1_slam_map_chunks_compaction_gate_admits_the_envelope.py` -- 24 passed

## Changelog

`changelog.d/PLACEHOLDER-g1-slam-map-chunks-compaction-envelope-port.md` -- will rename to `<PR#>-g1-slam-map-chunks-compaction-envelope-port.md` after this PR opens.

Refs #358.

---

## Review rounds

| Round | Concern | Fix commit | Pin test |
|---|---|---|---|
| R1 | Dead `:mod:` xref roles to in-flight sibling modules (#3026, #3027) block `test_docstring_xref_roles_resolve` | `ddebc66` | existing `test_docstring_xref_roles_resolve` (no new pin needed -- the concern is a CI-blocking test, not a behavioural regression) |
| R2 | #3026 merged at 20:52Z, four minutes before the R1 commit called it "in flight" -- a landed sibling described as unlanded in 4 module citations, 1 test citation and the changelog fragment | `32c5619` (absorb main), `f8e8940` | existing `test_docstring_xref_roles_resolve`; negative control confirms it reports these exact citations when a target is misspelled, so they are genuinely resolved rather than skipped |

### CODE REVIEW

File: strands_robots/tools/g1/g1_slam_map_chunks_compaction_envelope.py [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+long-running SLAM accumulation session can decide the chunk-list bound
+decidably before a future driver-side wrapper for the runner fires,
+rather than reading the neon runner's silent steal-and-dedup behaviour
+off a list that has already lost its per-chunk provenance.  A caller
+reading ``g1_slam_map_chunks_compaction_admits(len)`` with its intended
+batch depth sees the ``100``-entry ceiling and can either shorten the
+accumulation window, install its own compaction timer upstream of the
+runner, or accept the runner's amortised compaction semantics.
+
+Twin of :mod:`~strands_robots.tools.g1.g1_slam_pose_history_envelope`
```
Comment (yinsong1986) [inline strands_robots/tools/g1/g1_slam_map_chunks_compaction_envelope.py]: [MUST FIX] `:mod:`~strands_robots.tools.g1.g1_slam_pose_history_envelope`` and `:mod:`~strands_robots.tools.g1.g1_slam_frame_queue_envelope`` name modules that do not exist at this head -- #3026 and #3027 are still open, and both paths 404 at `ecb956be`. `tests/test_docstring_xref_roles_resolve.py::test_qualified_strands_robots_xref_roles_resolve` grades every qualified `:mod:` role in `strands_robots/` and `tests/` against the importable API, and running its resolver against this diff reports both targets as dead pointers: four occurrences in this module docstring, one in the `g1_slam_map_chunks_compaction_admits` docstring, and two more in the new test module's docstring. That test runs inside `call-test-lint / Test and Lint` -- the only required status check -- so this blocks the merge mechanically, and merging #3026/#3027 first does not clear it on its own: CI checks out the branch head, so the branch would still need to absorb `main` afterwards.

The fix is the convention this same PR's `g1_slam_icp_fitness_envelope.py` edit demonstrates: the text it *removes* explains that an in-flight sibling "is named as a literal rather than cross-referenced, because a Sphinx role promises an importable dotted path and this tree does not carry that module until #3006 lands." Apply that here -- spell the two in-flight twins as literal names (e.g. ``g1_slam_pose_history_envelope`` (strands-labs/robots#3026, in flight)) in both the module and test docstrings, and convert them to `:mod:` roles in a follow-up once #3026/#3027 merge, exactly as this PR does for #3006. (Separately, line 23 calls #3026 "the merged" while line 36 calls it "in flight" -- worth reconciling in the same edit, though that alone would not block.)

**Review by yinsong1986 — COMMENTED** [review]
## Summary

Ports the neon `_SlamRunner._process_frame` chunk-list compaction trigger (`len(self._map_chunks) > 100`) as a read-only lookup pair in `strands_robots.tools.g1.g1_slam_map_chunks_compaction_envelope`, following the established envelope-pair shape (list verb + admits grader on the shared `positive_count_error` domain, module-local refusal text rather than a borrowed motion rc, boundary semantics mirroring the runner's strict-greater check). Also converts the `g1_slam_icp_fitness_envelope` docstring's literal reference to `g1_slam_relocalize_envelope` into a `:mod:` role now that #3006 has landed, and ships the changelog fragment already named for this PR number. One blocker, flagged inline: the new module and test docstrings cite the two in-flight twins (#3026, #3027) with `:mod:` roles, and those modules do not exist at this head, which fails the repository's xref-resolution guard inside the sole required check.

## What's good

- The admits grader runs the shared-domain shape check before the ceiling comparison, refuses `bool`, and returns error dicts rather than raising -- consistent with the tool-surface conventions.
- `batch_depth` has a real `Args:` entry, so the derived tool schema carries a usable description (AGENTS.md convention 13).
- Boundary semantics (100 admits, 101 refuses) match the neon runner's `>` check exactly, and the test matrix pins both sides plus the shared-domain refusals.
- The `g1_slam_icp_fitness_envelope` edit is the correct literal-to-role conversion for the now-merged #3006 -- the exact convention the blocker inline comment asks this PR to apply to its own in-flight citations.
- Changelog fragment is named with the real PR number in the first push, so no fragment-race second push is needed.

File: strands_robots/tools/g1/g1_slam_map_chunks_compaction_envelope.py (hunk 1 above)
Comment (cagataycali) [reply strands_robots/tools/g1/g1_slam_map_chunks_compaction_envelope.py]: Fixed in `ddebc66` -- replaced every `:mod:` role targeting `g1_slam_pose_history_envelope` (4 occurrences in the source module, 2 in the test module) and `g1_slam_frame_queue_envelope` (1 source, 1 test) with literal backtick-quoted names plus the PR reference, matching the convention this same PR's `g1_slam_icp_fitness_envelope.py` edit demonstrates. The "merged #3026" vs "in flight" inconsistency on line 23 is also reconciled (now reads "in flight" consistently).

Remaining `:mod:` roles point only at modules that do exist on this branch: `g1_slam_voxel_dedup_envelope` (merged #3014) and `g1_slam_relocalize_envelope` (merged #3006). 24/24 tests pass.

---

> **AI Disclosure**: This comment was authored by an autonomous agent ([strands-agents](https://github.com/strands-agents)). The fix was reviewed against the codebase convention before pushing.

**Review by yinsong1986 — APPROVED** [review]
## Summary

Ports the neon `_SlamRunner._process_frame` chunk-list compaction trigger (`len(self._map_chunks) > 100`, steal-and-dedup on strict-greater) as a read-only lookup pair in `strands_robots.tools.g1.g1_slam_map_chunks_compaction_envelope`, following the envelope-pair template the merged siblings (#3006, #3014, #3026) established: a list verb returning the neon-observed constant plus a module-local refusal descriptor, and an admits grader that runs the shared `positive_count_error` domain check ahead of the ceiling comparison. The module is a leaf under `tools/g1/` (per the package's no-re-export convention), pulls no optional SLAM stack at import, keeps every tool-result string ASCII, documents its one `@tool` parameter under `Args:`, and correctly declines to borrow the motion-FSM `7404` rc for a bookkeeping refusal. Both prior-review concerns (dead `:mod:` roles to then-unlanded siblings) are resolved at this head -- `g1_slam_pose_history_envelope`, `g1_slam_voxel_dedup_envelope` and `g1_slam_relocalize_envelope` all exist at `f8e89401` after the branch absorbed main, and the remaining in-flight sibling is cited with literal backticks only. Changelog fragment is named with the real PR number.

## What's good

- Boundary matrix pins the runner's `>` comparison exactly: 100 admits, 101 refuses, and the shared-domain shape refusals (`0`, `-1`, `True`, `False`, float/str/None) are graded before the ceiling so the refusal `comparison` field is decidable.
- Import-hygiene tests pop the target module first so the graded delta is the module's own import cost, and the refusal text is asserted shared across both grades and the list verb.
- The twin-envelope separation rationale (compaction trigger vs dedup cell size vs pose history vs frame queue) is argued in the docstring with per-surface remedies, matching the merged siblings' convention.
